v3/mqtt.go

```diff
@@ -291,19 +291,31 @@ func NewMqttClient(mqttHost string, mqttPort int,
 // IsConnected will return true if we have a successfully CONNACK'ed response.
 //
 func (client *MqttClient) IsConnected() bool {
-	return client.conn != nil && client.conn.getStatus() == ConnectedNetworkStatus
+	return client.getConn() != nil && client.getConn().getStatus() == ConnectedNetworkStatus
```

Comment:
I think, if it is possible of race condition in this object,  this if statement needs to enclose by mutex. If it doesn't use mutex, it is possible to happen nil reference panic.

1. in goroutine A: `client.getConn() != nil # => true`
2. in goroutine B: `client.setConn(nil)`
3. in goroutine A: `client.getConn().getStatus() == ConnectedNetworkStatus # => panic`


```suggestion
	client.mtx.Lock()
	defer client.mtx.Unlock()
	return client.getConn() != nil && client.getConn().getStatus() == ConnectedNetworkStatus
```

Reply:
I think it exists similar situation in other methods too.

Reply:
But this suggestion code is multiple locks.

Reply:
Good point. You are right. It can be happen. There were no nil guard to `client.conn` in other functions too. It is another issue to be fixed. I will create a task to fix the issue.

Reply:
So... This PR be keep? Or will fix?

Reply:
I think it is okay to merge. The race condition issue is another topic. So we can create another pull request to solve the issue.
@takeshinoda is it okay for you to merge thiis pull request?

Reply:
I see! I can approve if so.